mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs

```diff
@@ -853,10 +855,233 @@ class PacketReader {
+					return ReadInt ();
+			}
+
+			public object ReadLongValue () {
+				var lenFixedSize = ReadFixedSize ();
+				if (lenFixedSize > 1)
+				{
+					var val = new long[lenFixedSize];
+					for (int i = 0; i < lenFixedSize; i++)
+					{
+							val[i] = ReadLong ();
+					}
+					return val;
+				}
+				else 
+					return ReadLong ();
+			}
+
+			public object ReadULongValue () {
+				var lenFixedSize = ReadFixedSize ();
+				if (lenFixedSize > 1)
+				{
+					var val = new ulong[lenFixedSize];
+					for (int i = 0; i < lenFixedSize; i++)
+					{
+							val[i] = (ulong) ReadLong ();
+					}
+					return val;
+				}
+				else 
+					return (ulong) ReadLong ();
+			}
+
+			public object ReadFloatValue () {
+				var lenFixedSize = ReadFixedSize ();
+				if (lenFixedSize > 1)
+				{
+					var val = new float[lenFixedSize];
+					for (int i = 0; i < lenFixedSize; i++)
+					{
+							val[i] = ReadFloat ();
+					}
+					return val;
+				}
+				else 
+					return ReadFloatValue ();
+			}
+
+
+			public object ReadDoubleValue () {
+				var lenFixedSize = ReadFixedSize ();
+				if (lenFixedSize > 1)
+				{
+					var val = new double[lenFixedSize];
+					for (int i = 0; i < lenFixedSize; i++)
+					{
+							val[i] = ReadDouble ();
+					}
+					return val;
+				}
+				else 
+					return ReadDouble ();
+			}
+			public ValueImpl ReadValueFixedSize () {
+				ElementType etype = (ElementType)ReadByte ();
+				switch (etype) {
+				case ElementType.Void:
+					return new ValueImpl { Type = etype };
+				case ElementType.I1:
+					return new ValueImpl { Type = etype, Value = ReadSbyteValue () };
+				case ElementType.U1:
+					return new ValueImpl { Type = etype, Value = ReadByteValue () };
+				case ElementType.Boolean:
+					return new ValueImpl { Type = etype, Value = ReadBolleanValue () };
+				case ElementType.I2:
+					return new ValueImpl { Type = etype, Value = ReadShortValue() };
+				case ElementType.U2:
+					return new ValueImpl { Type = etype, Value = ReadUShortValue () };
+				case ElementType.Char:
+					return new ValueImpl { Type = etype, Value = ReadCharValue () };
+				case ElementType.I4:
+					return new ValueImpl { Type = etype, Value = ReadIntValue () };
+				case ElementType.U4:
+					return new ValueImpl { Type = etype, Value = ReadUIntValue () };
+				case ElementType.I8:
+					return new ValueImpl { Type = etype, Value = ReadLongValue () };
+				case ElementType.U8:
+					return new ValueImpl { Type = etype, Value = ReadULongValue () };
+				case ElementType.R4:
+					return new ValueImpl { Type = etype, Value = ReadFloatValue () };
+				case ElementType.R8:
+					return new ValueImpl { Type = etype, Value = ReadDoubleValue () };
+				}
+				throw new NotImplementedException ("Unable to handle type " + etype);
+			}
 			public ValueImpl ReadValue () {
 				ElementType etype = (ElementType)ReadByte ();
-
 				switch (etype) {
+				case (ElementType)ValueTypeId.VALUE_TYPE_ID_FIXED_ARRAY:
+					return ReadValueFixedSize ();
```

Comment:
This could be simplified now since we know its a fixed array, no need for the Read<something>Value methods.